src/components/data-table-v2/data-table-v2.js

```diff
@@ -0,0 +1,310 @@
+      const eventElement = eventMatches(evt, this.options.selectorChildRow);
+
+      if (eventElement) {
+        this._expandableHoverToggle(eventElement, true);
+      }
+    });
+
+    this.element.addEventListener('click', evt => {
+      const eventElement = eventMatches(evt, this.options.eventTrigger);
+      if (eventElement) {
+        this._toggleState(eventElement, evt);
+      }
+    });
+
+    this.element.addEventListener('keydown', this._keydownHandler);
+
+    this.state = {
+      checkboxCount: 0,
+    };
+  }
+
+  _sortToggle = detail => {
+    const { element, previousValue } = detail;
+
+    this.tableHeaders.forEach(header => {
+      const sortEl = header.querySelector('.bx--table-sort-v2');
+
+      if ((sortEl !== null) & (sortEl !== element)) {
+        sortEl.classList.remove(this.options.classTableSortActive);
+        sortEl.classList.remove(this.options.classTableSortAscending);
+      }
+    });
+
+    if (!previousValue || previousValue === 'descending') {
+      element.dataset.previousValue = 'ascending';
+      element.classList.add(this.options.classTableSortActive);
+      element.classList.add(this.options.classTableSortAscending);
+    } else {
+      element.dataset.previousValue = 'descending';
+      element.classList.add(this.options.classTableSortActive);
+      element.classList.remove(this.options.classTableSortAscending);
+    }
+  };
+
+  _selectToggle = detail => {
+    const { element } = detail;
+    const checked = element.checked;
+
+    // increment the  count
+    this.state.checkboxCount += checked ? 1 : -1;
+    this.countEl.textContent = this.state.checkboxCount;
+
+    // toggle on/off all-checkbox if all are selected
+
+    // toggle on/off batch action bar
+    this._actionBarToggle(this.state.checkboxCount > 0);
+  };
+
+  _selectAllToggle = detail => {
+    const checked = detail.element.checked;
+
+    const inputs = [...this.element.querySelectorAll(this.options.selectorCheckbox)];
+
+    this.state.checkboxCount = checked ? inputs.length - 1 : 0;
+
+    inputs.forEach(item => {
+      item.checked = checked;
+    });
+
+    this._actionBarToggle(this.state.checkboxCount > 0);
+    this.countEl.textContent = this.state.checkboxCount;
+  };
+
+  _actionBarCancel = () => {
+    const inputs = [...this.element.querySelectorAll(this.options.selectorCheckbox)];
+
+    inputs.forEach(item => {
+      item.checked = false;
+    });
+
+    this.state.checkboxCount = 0;
+    this._actionBarToggle(false);
+    this.countEl.textContent = this.state.checkboxCount;
+  };
+
+  _actionBarToggle = toggleOn => {
+    const transition = evt => {
+      this.batchActionEl.removeEventListener('transitionend', transition);
+
+      if (evt.target.matches(this.options.selectorActions)) {
+        if (this.batchActionEl.dataset.active === 'false') {
+          this.batchActionEl.setAttribute('tabIndex', -1);
+        } else {
+          this.toolbarEl.style.visibility = 'hidden';
```

Comment:
Good to have a CSS class for this visibility state.

Reply:
Turns out the visibility state isn't required, so removing it